packages/flutter_tools/test/general.shard/build_system/targets/assets_test.dart

```diff
@@ -461,6 +461,162 @@ flutter:
     },
   );
 
+  group('platform-specific assets', () {
+    /// All supported platforms that should be validated.
+    const kValidPluginPlatforms = <String>{'android', 'ios', 'web', 'windows', 'linux', 'macos'};
+
+    TargetPlatform targetFor(String platform) =>
+        TargetPlatform.values.firstWhere((p) => p.osName == platform);
+
+    /// Writes a `pubspec.yaml` with an asset, optionally restricted to
+    /// certain [platforms], then runs the build for [targetPlatform] and
+    /// returns whether the asset was bundled.
+    ///
+    /// This helper reflects how Flutter decides which assets to include
+    /// depending on the `platforms:` key in `pubspec.yaml`.
+    Future<bool> setupAndBuildPlatformAsset(String platform, TargetPlatform targetPlatform) async {
+      final filePath = 'assets/test-$platform.txt';
+
+      final pubspec = platform.isEmpty
+          ? '''
+name: example
+flutter:
+  assets:
+    - path: $filePath
+'''
+          : '''
+name: example
+flutter:
+  assets:
+    - path: $filePath
+      platforms:
+        - $platform
+''';
+
+      fileSystem.file('pubspec.yaml')
+        ..createSync()
+        ..writeAsStringSync(pubspec);
+      writePackageConfigFiles(directory: globals.fs.currentDirectory, mainLibName: 'example');
+      fileSystem.file(filePath).createSync(recursive: true);
+
+      await const CopyAssets().build(environment, targetPlatform: targetPlatform);
+
+      final File file = fileSystem.file('${environment.buildDir.path}/flutter_assets/$filePath');
+      final bool exists = file.existsSync();
+      if (exists) {
+        file.deleteSync();
+      }
+
+      return exists;
+    }
+
+    testUsingContext(
```

Comment:
Prefer making this a `group` which uses the loop to define tests for each individual platform. That way we don't need to include the delete logic above as the file system will be destroyed between tests and won't leak state across tests for individual platforms.